File: scala_examples/makeAdder.scala
/**
 * Creating a function that increments from a given (closed-over) starting value, using a closure
 */

object ClosureEx extends App {
  def makeAdder(x: Int): (Int) => Int = {
    def inc(y: Int): Int = {
      x + y
    }
    inc
  }
  
  val inc5 = makeAdder(5)
  val inc10 = makeAdder(10)
  
  println(inc5(2))  // 7
  println(inc10(2)) // 12
    
}      

File: python_examples/makeAdder.py
## CreatingCreating a function that increments from a given (closed-over) starting value, using a closure
def makeAdder(x):
    def inc(y):
        return x+y
    return inc

inc5  = makeAdder(5)
inc10 = makeAdder(10)

print(inc5 (2)) # 7
print(inc10(2)) # 12